nomad/structs/structs.go

```diff
@@ -316,12 +317,20 @@ type NodeRegisterRequest struct {
 }
 
 // NodeDeregisterRequest is used for Node.Deregister endpoint
-// to deregister a node as being a schedulable entity.
+// to deregister a batch of nodes from being schedulable entities.
+// Deprecated in 0.9.3
```

Comment:
Given that we still use it in http agent nodePurge, we might want to qualify its deprecation and scope it to internal scheduler work.